crates/valence_inventory/src/lib.rs

```diff
@@ -397,6 +399,8 @@ impl ClientInventoryState {
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Component, Deref)]
 pub struct HeldItem {
     held_item_slot: u16,
+    #[deref(ignore)]
+    changed: bool,
```

Comment:
Can't we use bevy's built-in change detection instead?
https://docs.rs/bevy/0.9.1/bevy/ecs/query/struct.Changed.html

Reply:
no because when the client change slot, we trigger one.

Reply:
And that's where I wanted to go with my review, what I proposing is that the system, who handle the packet bypass the change detection, so the only way the user of the api know that something occurs would be the event send.
now you would'nt need a seconde variable.

Reply:
I dont really understand what needs to be bypassed, as i mentioned, this isnt triggered from something automatically, this function gives a developer the option to change a clients selected slot index. Imo, this is simple enough to use as-is, and when https://github.com/valence-rs/valence/pull/487 gets finalized, this can be look at again if required?

Reply:
I'm talking about the system that handle the c2s held_item packet, the things I don't like here is the fact that you need a double change detection

Reply:
i havent touched anything related to c2s in this pr, out of scope?

Reply:
no because, how the c2s was implemented was making sense when you didn't care that the change detection was trigger, but now you care for your purpose.